### PR TITLE
[huggingface_tensorflow, huggingface_pytorch] update for Transformers to 4.10

### PR DESCRIPTION
**Pull request by philschmid**
.. 4.9.2

*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description

This PRs updates the `transformers` and `datasets` versions inside the huggingface PyTorch and TensorFlow DLC.

### DLC image/dockerfile

_Change only in `buildspec.yaml` required._

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [x] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [x] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true` or `neuron_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by philschmid** [comment]
Needs https://github.com/huggingface/transformers/pull/13181 to be merged.

**Comment by arjkesh** [comment]
PT 1.7 won't be built/ttested by default in this PR. In order to test PT1.7, you'll need to swamp out the buildspec with the PT1.7 buildspec in a revision.

**Review by arjkesh — APPROVED** [review]

**Review by ahsan-z-khan — DISMISSED** [review]

**Review by ahsan-z-khan — APPROVED** [review]

File: huggingface/tensorflow/inference/docker/2.4/py3/cu110/Dockerfile.gpu [hunk 1]
```diff
@@ -12,7 +12,7 @@ ARG PYTHON=python3
 ARG PYTHON_VERSION=3.7.10
 ARG OPEN_MPI_VERSION=4.0.1
 # HF ARGS
-ARG TF_URL=https://aws-tensorflow-binaries.s3-us-west-2.amazonaws.com/tensorflow/r2.4_aws/20210127-150238/gpu/py37/cu110/tensorflow_gpu-2.4.1-cp37-cp37m-manylinux2010_x86_64.whl
+ARG TF_URL=https://aws-tensorflow-binaries.s3-us-west-2.amazonaws.com/tensorflow/r2.4_aws/20210127-150238/gpu/py37/cu110/tensorflow_gpu-2.4.3-cp37-cp37m-manylinux2010_x86_64.whl
```
Comment (arjkesh) [inline huggingface/tensorflow/inference/docker/2.4/py3/cu110/Dockerfile.gpu:15]: I don't think these links will be valid - do you just generally use the one from the tensorflow DLC? i.e. here? https://github.com/aws/deep-learning-containers/blob/master/tensorflow/training/docker/2.4/py3/cu110/Dockerfile.gpu#L37

File: huggingface/tensorflow/inference/docker/2.4/py3/Dockerfile.cpu [hunk 2]
```diff
@@ -15,7 +15,7 @@ ARG PYTHON=python3
 ARG PYTHON_VERSION=3.7.10
 ARG OPEN_MPI_VERSION=4.0.1
 # HF ARGS
-ARG TF_URL=https://aws-tensorflow-binaries.s3-us-west-2.amazonaws.com/tensorflow/r2.4_aws/20210127-150238/cpu/py37/tensorflow_cpu-2.4.1-cp37-cp37m-manylinux2010_x86_64.whl
+ARG TF_URL=https://aws-tensorflow-binaries.s3-us-west-2.amazonaws.com/tensorflow/r2.4_aws/20210127-150238/cpu/py37/tensorflow_cpu-2.4.3-cp37-cp37m-manylinux2010_x86_64.whl
```
Comment (arjkesh) [inline huggingface/tensorflow/inference/docker/2.4/py3/Dockerfile.cpu:18]: I don't think these links will be valid - do you generally use the one from the tensorflow DLC? i.e. here? https://github.com/aws/deep-learning-containers/blob/master/tensorflow/training/docker/2.4/py3/Dockerfile.cpu#L30

File: huggingface/tensorflow/buildspec.yml [hunk 3]
```diff
@@ -30,14 +30,14 @@ images:
   BuildHuggingFaceTensorflowGpuPy37Cu110TrainingDockerImage:
     <<: *TRAINING_REPOSITORY
     build: &HUGGINGFACE_TENSORFLOW_GPU_TRAINING_PY3 false
-    image_size_baseline: &IMAGE_SIZE_BASELINE 8489
+    image_size_baseline: &IMAGE_SIZE_BASELINE 10500
```
Comment (arjkesh) [inline huggingface/tensorflow/buildspec.yml:33]: It seems that the base image caused this to get a lot bigger - how large are the packages installed on top of the DLC? @ahsan-z-khan - is this too big for SM?

Comment (philschmid) [reply huggingface/tensorflow/buildspec.yml:33]: We additionally only installing `transformers` and 3 more python packages they should be < 200MB

File: test/sagemaker_tests/huggingface/inference/integration/sagemaker/test_hub_model.py [hunk 4]
```diff
@@ -29,32 +26,32 @@
...
+        dump_logs_from_cloudwatch(e, region)
         raise
 
 
 def _test_hub_model(sagemaker_session, framework_version, ecr_image, instance_type, model_dir, accelerator_type=None):
     endpoint_name = sagemaker.utils.unique_name_from_base("sagemaker-huggingface-serving-hub-model")
 
     env = {
-        "HF_MODEL_ID": "sshleifer/tiny-distilbert-base-uncased-finetuned-sst-2-english",
+        "HF_MODEL_ID": "philschmid/tiny-distilbert-classification",
```
Comment (arjkesh) [inline test/sagemaker_tests/huggingface/inference/integration/sagemaker/test_hub_model.py:54]: Is there a way we could remove GitHub ID from this and use something more generic?

Comment (philschmid) [reply test/sagemaker_tests/huggingface/inference/integration/sagemaker/test_hub_model.py:54]: What do you mean with Github ID and generic? It is a repository on hf.co/models https://huggingface.co/philschmid/tiny-distilbert-classification, which is maintained by me and which should be tested in the test

**Review by arjkesh — APPROVED** [review]

File: test/dlc_tests/sanity/test_pre_release.py [hunk 5]
```diff
@@ -345,6 +343,9 @@ def test_dependency_check_cpu(cpu, ec2_connection):
     reason="Executing test in canaries pipeline during only a limited period of time.",
 )
 def test_dependency_check_gpu(gpu, ec2_connection):
+    # TODO: Fix test on HF inference
+    if "huggingface-tensorflow-inference" in gpu or "huggingface-pytorch-inference" in gpu:
```
Comment (shantanutrip) [inline test/dlc_tests/sanity/test_pre_release.py:347]: Are we planning to merge the PR with these tests skipped?

Comment (arjkesh) [reply test/dlc_tests/sanity/test_pre_release.py:347]: Yeah, I've updated the corresponding release GitHub issue to state that we need to run this test manually for release. I've tested manually and the test passes, so we'll need to follow up separately to dive into the infrastructure issue causing this failure

**Review by ahsan-z-khan — APPROVED** [review]

**Pull request merged**